### DISCUSSION TITLE
Add html pages to Nextjs

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#21226] [Help]
Hi

I want to know where should I place HTML pages inside nextjs? And how to route to those pages?

thank you

**Comment by vassbence** [comment]
Next.js is a React framework, while you definitely can place arbitrary (so even  *.html) files in [the public folder](https://nextjs.org/docs/basic-features/static-file-serving) for Next.js to serve as static content, if we are talking about just a regular page of a website [you should create a *.js file in the pages folder.](https://nextjs.org/docs/basic-features/pages) Next will take care of routing based on the file name you give (read more in the previous link about the details). If you have a simple static page html file it will not be hard to "migrate" it to JSX/React.

Just do this in for example `pages/index.js`

```js
export default function Page() {
  return (__your html here__)
}
```

Make sure to not include`<html>` tags and for anything that would normally go into the `<head>` use [next/head](https://nextjs.org/docs/api-reference/next/head). So basically take your html starting form after the `<body>` and copy it into the function I wrote above :)

**Reply by ghost** [reply]
thank you @vassbence I have a component that will be served from Next js pages but I have also static pages created by other member need to place it in the project and connect it to the routes so I dont need to change anything in those pages it has its own css folder and js etc.
I just need to connect them to the routes for example 
I had a similar thing in CRA but I am migrating now 
`<Route path="/home" render={() => {window.location.href="home.html"}}`
so that is all i had to do in CRA

**Reply by vassbence** [reply]
Take a look at [rewrites](https://nextjs.org/docs/api-reference/next.config.js/rewrites).

You should be able to place the `home.html` file in `public/static/home.html` for example and have a rewrite from `/home` to `/static/home.html`

**Reply by ghost** [reply]
Is it necessary to create static folder in public? if this method will work i will place the HTML pages with their styling and js files in the public folder and access those pages with this path `/home` for example
